ckan/logic/action/create.py

```diff
@@ -296,6 +296,12 @@ def resource_create(context, data_dict):
 
     upload = uploader.get_resource_uploader(data_dict)
 
+    # determine size for the file to upload
+    if upload.filename:
+        upload.upload_file.seek(0, 2)
```

Comment:
need to catch possible IOError here and leave the size unchanged for file objects passed that don't support seek.